lib/src/kbasesearchengine/common/GUID.java

```diff
@@ -46,6 +60,16 @@ public GUID(final GUID parentGUID, final String subObjectType, final String subO
         version = parentGUID.getVersion();
         this.subObjectType = subObjectType;
         this.subObjectId = subObjectID;
+
+        validate();
+    }
+
+    private void validate() {
+        final int guidBytes = toString().getBytes(StandardCharsets.UTF_8).length;
```

Comment:
The other option for really long GUIDs is to hash/uuidify the subobject ID (which is the only part that can be really long, most likely). 

I'm ok with throwing an exception as well. 

@scanon do you have a preference? If not let's go with the exception and try reindexing CI and see if we have any problems. Need to make sure we're indexing subobjects.

Reply:
Mentioned this to @scanon and he's fine with an exception, but the exception should be specific so that we can search the DB for the exception type to see how many of these we get.

Reply:
Done.

Reply:
Anything that's a RuntimeException is not going to wind up in the recorded in the db, it's just going to kill the thread.

Reply:
I turned it into a runtime exception out of laziness. This code is used in over 500 places. I can make the updates to link the checked exceptions if you went. It's going to be a lot of updates and I'm not sure it's worth it for a rare case

Reply:
The point of the special exception was to be able to check the DB and see if we have any problems with long IDs when we index CI, so switching the exception to a Runtime is kind of pointless. Maybe check the length issue outside the GUID class prior to the load in a part of the code that already throws IndexingExceptions?

Reply:
That would defeat the guid validation. I'll give the checked exception linking a crack and see if I can wing it without it getting out of hand

Reply:
Or just catch and wrap the unchecked exception at the appropriate place and document the unchecked exception.

Reply:
Same thing right? I'd have to cover for over 500 possible places where this exception would need to be caught.

Reply:
I don't think so, just catching it here would work:
https://github.com/kbase/KBaseSearchEngine/blob/develop/lib/src/kbasesearchengine/main/IndexerWorker.java#L650
All the sub object IDs are created in there I think.

Reply:
I gave it a try. It is getting really ugly. Here are the files I have modified and I am still not done. I feel the unchecked exception is the best bet here. It's highly unlikely we are going to run into a guid that is 500 chars long - a guid that is 5 lines long. 

```
	modified:   lib/src/kbasesearchengine/common/GUID.java
	modified:   lib/src/kbasesearchengine/common/GUIDTooLongException.java
	modified:   lib/src/kbasesearchengine/events/StatusEvent.java
	modified:   lib/src/kbasesearchengine/events/exceptions/IndexingException.java
	modified:   lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java
	modified:   lib/src/kbasesearchengine/main/IndexerCoordinator.java
	modified:   lib/src/kbasesearchengine/main/IndexerWorker.java
	modified:   lib/src/kbasesearchengine/parse/KeywordParser.java
	modified:   lib/src/kbasesearchengine/parse/ObjectParser.java
	modified:   lib/src/kbasesearchengine/search/ElasticIndexingStorage.java
	modified:   lib/src/kbasesearchengine/search/ObjectData.java
	modified:   test/src/kbasesearchengine/test/common/GUIDTest.java
	modified:   test/src/kbasesearchengine/test/events/StatusEventTest.java
	modified:   test/src/kbasesearchengine/test/events/handler/ResolvedReferenceTest.java
	modified:   test/src/kbasesearchengine/test/parse/KeyWordParserTest.java

```

Reply:
I think our messages crossed. I have added a check for the unchecked exception as you suggested.